### ISSUE TITLE
Simple question: Firefox showing different results than major browsers

### ISSUE THREAD

**Issue by Lepefe** [d3ward/toolz#50]
Hi @d3ward ,

Firstly, thank you for your Toolz page.

I did a test with your page, to compare ads blocked, between Firefox and Google Chrome.
I used 3 different devices, different days, and I used vanilla versions of FF and GC (at my test, both browsers were untouched, no add-ons, no extensions, latest versions etc).

In my test, and as expected, GC blocked few ads (around 3% on your test page).
However, unexpectedly, FF blocked around 75% (on your test page). So my next step was to try to understand the reason.

Step by step, I disabled each FF privacy setting. Then, security settings. Then, I enabled cookies. And finally I changed lot of about:config settings in order to disable, as much as possible, all the FF blocking capabilities. Guess what? At your test page, FF kept blocking around 75% of the ads.

Then I opened a discussion at Reddit/Firefox (https://old.reddit.com/r/firefox/comments/11dbsjj/testing_ff_gchrome_please_i_need_an_explanation/).
IMHO, the 75% blocked stuff is a FF bug.
But FF fanboys don't like critics (they even down-voted my post and my comments). And they blamed your page.
In short, no answers and no understanding at Reddit/Firefox.

Now my intention is to open a bug report at Bugzilla.
Before that, please, I would like your opinion.
Basically my question is, at your test page, how FF can block around 75% of the stuff, when FF blocking capabilities are disabled?

Thank you in advance!

**Comment by d3ward** [comment]
Mmmm. do you use any dns that is different from the default one provided by your ISP ? There are some dns that have blocking capabilities and i remember a couple of them got around 75%

**Comment by Lepefe** [comment]
> Mmmm. do you use any dns that is different from the default one provided by your ISP ? There are some dns that have blocking capabilities and i remember a couple of them got around 75%

Thank you for your replay.

Nope.
For test purpose, I only used automatic DNS and Google DNS.
And the DNS was not the problem in my test because I also tested other major browsers (Google Chrome, Vivaldi, Edge, Safari, Opera etc)... and for these browsers your test page shows that most of the ads are unblocked (around 3%).

Firefox is the only browser in your test page, showing around 75% of stuff blocked, while most of the Firefox blocking capabilities are disabled.

It might be an incompatibility between Firefox and your test page, or it might be a Firefox bug.

**Comment by d3ward** [comment]
I see . Ok , i will start doing some tests with Firefox . I did test it on MacOS and didn't have any issue , worked as expected. I guess you are on Windows so will look into that . Thanks !

**Comment by Lepefe** [comment]
Thank you for look at it. Please, I'll wait for your results. If you don't find issues with your test page, then it will mean that the problem is Firefox (and I'll report it at Bugzilla as a Firefox bug).

And yes, I'm using Windows 11, latest version, fully updated.

**Comment by bloodyeyesblackblood** [comment]
https://github.com/d3ward/toolz/issues/53#issuecomment-1528812729
> maybe the problem here is "Enhanced Tracking Protection"?
at least my score is lower when it's also enabled.

do you have it enabled? if yes, what is your score when you disable it?

**Comment by Lepefe** [comment]
In my case, as I wrote long time ago:

> Step by step, I disabled each FF privacy setting. Then, security settings. Then, I enabled cookies. And finally I changed lot of about:config settings in order to disable, as much as possible, all the FF blocking capabilities. Guess what? At your test page, FF kept blocking around 75% of the ads.
>

**Comment by d3ward** [comment]
@Lepefe I know. It's a complex thing. I'm considering a couple of options there, either shutting down the project or banning Firefox users.... I'm not sure how to fix this annoying problem.

**Comment by Lepefe** [comment]
> @Lepefe I know. It's a complex thing. I'm considering a couple of options there, either shutting down the project or banning Firefox users.... I'm not sure how to fix this annoying problem.

One thing is for sure: Your test-webpage behaves different in Firefox than in other browsers.
IMHO, , the different behavior is not related to UBlock.
Unfortunately, I don't have know-how to help you. But your project is very useful!, so one day in the future (don't rush!), I hope to see it working in Firefox. Again, don't rush, take all the time you need.

Thank you for your effort!

**Comment by d3ward** [comment]
I have found that on Firefox the problem can be fixed with a specific rule. It's not directly ublock's fault, it's just the way it works on firefox is different by default to get the best performance/configuration.
Added Firefox to the "?" with a note for users. Thanks for the support and discussion.
